types/utxo.go

```diff
@@ -50,7 +50,7 @@ func (utxo *BaseUTXO) GetAddress() common.Address {
 
 //Implements UTXO
 func (utxo *BaseUTXO) SetAddress(addr common.Address) error {
-	if utils.ZeroAddress(utxo.Address) {
+	if !utils.ZeroAddress(utxo.Address) {
```

Comment:
Good catch..did we not have tests checking this before?

Reply:
no, still missing tests for keeper also